src/Routers/AggregateRouter.js

```diff
@@ -106,7 +106,8 @@ export class AggregateRouter extends ClassesRouter {
       stage[stageName]._id = stage[stageName].objectId;
       delete stage[stageName].objectId;
     }
-    return { [`$${stageName}`]: stage[stageName] };
+    const key = stageName[0] === '$' ? stageName : `$${stageName}`;
```

Comment:
Depending on how you want to proceed, remove feature of adding a dollar sign (without deprecation) or add the deprecation warning (with deprecation).

Reply:
Thanks for the guidance. I had a look at the [Deprecator](https://github.com/parse-community/parse-server/blob/master/src/Deprecator/Deprecator.js#L35). But if I'm getting it correctly, I would need to add the list of currently available aggregation stages in latest MongoDB 4.x, right? But that is a dynamic list, so if more stages get added, they won't be caught as a deprecation warning.

In this case, I think it would be better to test just for the missing `$`. If needed I can make the changes in Deprecator for that but I doubt it may be used in another case.

Any thoughts?

Reply:
> In this case, I think it would be better to test just for the missing $

Yes, makes sense.

> If needed I can make the changes in Deprecator for that but I doubt it may be used in another case.

I already expected that we'll need to add this to the deprecator, but I was waiting for the first case to implement this - and here it is. The concept of the deprecator is to define deprecations centrally, which is easy deprecating for Parse Server Options. For deprecations that are detected "in-code" I am still thinking what the best approach is. I would want to avoid spreading deprecation definitions all over the place.